### PR TITLE
New WikiHow IA

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

A fathead for wikihow.

screen shot 2017-07-20 at 11 06 16 pm

## Todo

- [ ] Discuss how many list items to display
- [ ] Deploy to beta or private DDG employee box
- [ ] Get internal feedback, update parser 

## People to notify

@moollaza @bsstoner @b2ddg 

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)

**Code**
- [ ] Uniformly indented, well commented
- [ ] Fetch.sh and Parse.sh run without errors
- [ ] Output contains no blank lines, or multi-line entries
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/wikihow

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/wikihow/fetch.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        tmp_error["reason"] = "bad response"
+
+    if bool(tmp_error):
+        fetch_errors.append(tmp_error)
+
+    seed_iterator += 1
+    time.sleep(.1) # slow me down
+
+# Finish up and create errors/warnings report
+print("Fin.")
```
Comment (TiagoDanin) [inline lib/fathead/wikihow/fetch.py]: Did you mean: "Finish." ?

File: lib/fathead/wikihow/output.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+How to "Bone" a Holster to Fit Your Handgun	A									[[Image:http://www.wikihow.com/images/a/a7/"Bone"-a-Holster-to-Fit-Your-Handgun-Step-10.jpg]]	<ol><li>Remove the contents from the packaging.</li><li>Obtain a suitably sized cooking pot from your cupboard and fill it with enough water to completely submerge the holster.</li><li>Prepare your handgun by applying plastic wrap over the entire weapon and use tape to secure it in place.</li><li>Apply the sight clearance tube over the front sight and align it along the center of the top to the rear sight and tape into place.</li><li>Apply another layer of plastic wrap and tape into place.</li><li>By now the water should be boiling.</li><li>Being careful of the heat of the holster, insert the handgun with the stitching centered on the trigger guard</li><li>Using the 'boning' tool, held at a 15-20 degree angle from the leather and applying firm pressure, begin to work the leather along the lines of your handgun.</li><li>Continue to work the leather to the handgun until you are satisfied with your creation.</li><li>Remove the plastic wrap and sight clearance tube.</li></ol>	http://www.wikihow.com/%22Bone%22-a-Holster-to-Fit-Your-Handgun
```
Comment (TiagoDanin) [inline lib/fathead/wikihow/output.txt]: Available HTTPS

Comment (kirkins) [inline lib/fathead/wikihow/output.txt]: @TiagoDanin it's actually not available in HTTPS, HTTPS on that link forwards to HTTP:

untitled

File: lib/fathead/wikihow/fetch.py (hunk 1 above)
Comment (kirkins) [inline lib/fathead/wikihow/fetch.py]: @TiagoDanin I think this is just a tongue and cheek reference to the french spelling of finish.

https://duckduckgo.com/?q=le+fin&t=h_&iax=1&ia=images

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

**Comment by daxtheduck** [comment]
@pjhampton this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**